deps/v8/src/sampler.cc

```diff
@@ -342,6 +342,14 @@ void SignalHandler::HandleProfilerSignal(int signal, siginfo_t* info,
     // We require a fully initialized and entered isolate.
     return;
   }
+
+  // ISSUE-14576: To avoid deadlock, return if there is a thread lock
+  if (Isolate::GetProcessWideMutexPointer()->TryLock()) {
```

Comment:
@dmelikyan @tunniclm Also, I'm still not sure I properly this fix. What is already locking the process wide mutex in the profiler thread when `isolate->thread_manager()->IsLockedByCurrentThread()` is called? Also, why is `Isolate::GetProcessWideMutexPointer()->TryLock()` returning true in case the process wide mutex is already locked? Shouldn't it return false since `man pthread_mutex_try_lock` says that `If the mutex is already locked, pthread_mutex_trylock() will not block waiting for the mutex, but will return an error condition`?